clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ProducerExample.java

```diff
@@ -0,0 +1,122 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.common.errors.TopicExistsException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.Collections;
+import java.util.Map;
+
+public class ProducerExample {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  // Create topic in Confluent Cloud
+  public static void createTopic(final String topic,
+                          final int partitions,
+                          final int replication,
+                          final Properties cloudConfig) {
+      final NewTopic newTopic = new NewTopic(topic, partitions, (short) replication);
+      try (final AdminClient adminClient = AdminClient.create(cloudConfig)) {
+          adminClient.createTopics(Collections.singletonList(newTopic)).all().get();
+      } catch (final InterruptedException | ExecutionException e) {
+          // Ignore if TopicExistsException, which may be valid if topic exists
+          if (!(e.getCause() instanceof TopicExistsException)) {
+              throw new RuntimeException(e);
+          }
+      }
+  }
+
+  public static void main(String[] args) throws IOException {
+    if (args.length != 2) {
+      System.out.println("Please provide command line arguments: configPath topic");
+      System.exit(-1);
+    }
+
+    // Load properties from file
+    Properties props = loadConfig(args[0]);
+
+    // Create topic if needed
+    String topic = args[1];
+    createTopic(topic, 1, 3, props);
+
+    // Add additional properties.
+    props.put(ProducerConfig.ACKS_CONFIG, "all");
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonSerializer");
+
+    Producer<String, JsonNode> producer = new KafkaProducer<String, JsonNode>(props);
+
+    // Produce sample data
+    final Long numMessages = 10L;
+    for (Long i = 0L; i < numMessages; i++) {
+      String key = "alice";
+      RecordJSON countRecord = new RecordJSON(i);
+      JsonNode value = MAPPER.valueToTree(countRecord);
+
+      producer.send(new ProducerRecord<String, JsonNode>(topic, key, value), new Callback() {
+          @Override
+          public void onCompletion(RecordMetadata m, Exception e) {
+            if (e != null) {
+              e.printStackTrace();
+            } else {
+              System.out.printf("Produced record to topic %s partition [%d] @ offset %d%n", m.topic(), m.partition(), m.offset());
+            }
+          }
+      });
+    }
+
+    producer.flush();
+
+    System.out.printf("10 messages were successfully produced to topic %s%n", topic);
```

Comment:
Since onCompletion neither fails or counts errors, this printout could be a lie.

Reply:
This is actually the same message in all our code examples, even the python one: 

https://github.com/confluentinc/examples/blob/5.0.1-post/clients/cloud/python/producer.py#L96

I understand your point, but IMO it's ok.  As long as the callback prints the error if there is one (which the callback does)

Reply:
You're asking me to be consistent in my reviews?!

All we need to do is remove "successfully" for this printout to be true, regardless of delivery status.

Reply:
Removed successfully.